workflows/data_pipelines/spectacle/spectacle_processor.py

```diff
@@ -0,0 +1,53 @@
+from dag_datalake_sirene.helpers import DataProcessor, Notification
+
+
+class SpectacleProcessor(DataProcessor):
+    def __init__(self):
+        from dag_datalake_sirene.workflows.data_pipelines.spectacle.spectacle_config import (
+            SPECTACLE_CONFIG,
+        )
+
+        super().__init__(SPECTACLE_CONFIG)
+
+    def preprocess_data(self):
+        import pandas as pd
+        from airflow.operators.python import get_current_context
```

Comment:
I know we discussed imports briefly, and I’m still not entirely convinced that there would be a huge memory gain since we haven’t profiled the code.

I prefer following PEP 8’s recommendation of keeping imports at the top of the file for better readability. While it might feel like it creates some clutter, that could just be because I’m used to having them at the top. Ultimately, if a method is called multiple times, having imports inside it could lead to repeated imports, causing the performance issues we’re trying to avoid. Plus, if an import is missing, we might not realize it until the function is executed, which isn’t ideal.

That said, I don’t think we’re doing heavy enough processing for this to make a noticeable difference either way. My main concern is consistency. If we decide to move imports, it should be done across all files to maintain uniformity.

What do you think?

Reply:
I am fine with it, I may sometimes do that again since I am used to do it so don't mind making a comment if it happens 😅 
I used to work on a repo where the Airflow parsing would take almost a full minute slowing everything down in the process haha. We ended up making the "parsing" kind of manual in prod but in dev it was annoying so we went with the code optimisation route.